### PR TITLE
Harnesses for `count_bytes` 

### LINKED ISSUE #150: Challenge 13: Safety of `CStr`
Challenge link: TBD

### PR DESCRIPTION
**Pull request by MWDZ**
Towards #150 

Changes
Added harnesses for count_bytes

Verification Result
```
Checking harness ffi::c_str::verify::check_count_bytes...

VERIFICATION RESULT:
 ** 0 of 241 failed (5 unreachable)

VERIFICATION:- SUCCESSFUL
Verification Time: 5.377671s
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: library/core/src/ffi/c_str.rs [hunk 1]
```diff
@@ -875,4 +875,28 @@ mod verify {
...
         }
     }
+
+    #[kani::proof]
+    #[kani::unwind(32)]
+    fn check_count_bytes() {
+        const MAX_SIZE: usize = 32;
+        let mut string: [u8; MAX_SIZE] = kani::any();
+        
+        // Randomly generate a length within the valid range [0, MAX_SIZE]
```
Comment (tautschnig) [inline library/core/src/ffi/c_str.rs:885]: I'd say "Non-deterministically" as people may associate some form of (normal) distribution with "random".

**Review by tautschnig — APPROVED** [review]

File: library/core/src/ffi/c_str.rs [hunk 2]
```diff
@@ -875,4 +875,28 @@ mod verify {
             assert!(c_str.is_safe());
         }
     }
+
+    #[kani::proof]
+    #[kani::unwind(32)]
+    fn check_count_bytes() {
+        const MAX_SIZE: usize = 32;
+        let mut string: [u8; MAX_SIZE] = kani::any();
```
Comment (celinval) [inline library/core/src/ffi/c_str.rs:883]: Can you rename this to `bytes`?

**Review by celinval — APPROVED** [review]
Thanks

File: library/core/src/ffi/c_str.rs (hunk 1 above)
Comment (MWDZ) [reply library/core/src/ffi/c_str.rs]: Thank you for the clarification! I've updated the comment to use "non-deterministically" to better reflect the behavior of `kani::any()`

File: library/core/src/ffi/c_str.rs (hunk 2 above)
Comment (MWDZ) [reply library/core/src/ffi/c_str.rs]: Thank you for pointing that out! I've renamed it to `bytes` to better reflect its purpose before it becomes a string.

**Pull request merged**